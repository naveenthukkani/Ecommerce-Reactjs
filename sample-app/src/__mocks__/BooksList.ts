import {Books} from "../../src/home/Books";

export const BooksList: Array<Books> = [
   {
       name:"maths",
       discription: "it is simple problems book for children",
       author:"naveen",
       price: 5,
       pagecount: 1,
	   isbn:"",
	   id:1,
   },
   {
    name:"social",
    discription: "it is sample book for children",
    author:"naveen",
    price: 7,
    pagecount: 1,
	isbn:"",
	id:2,
},
{
    name:"telugu",
    discription: "it is language learning book",
    author:"naveen",
    price: 8,
    pagecount: 1,
	isbn:"",
	id:3,
},
{
    name:"English",
    discription: "it is international language learning book",
    author:"naveen",
    price: 5,
    pagecount: 1,
	isbn:"",
	id:4,
},
{
    name:"maths",
    discription: "it is simple problems book for children",
    author:"naveen",
    price: 5,
    pagecount: 1,
	isbn:"",
	id:5,
},
{
	name:"social",
	discription: "it is sample book for children",
	author:"naveen",
	price: 7,
	pagecount: 1,
	isbn:"",
	id:6,
},
{
	name:"telugu",
	discription: "it is language learning book",
	author:"naveen",
	price: 8,
	pagecount: 1,
	isbn:"",
	id:7,
},
{
	name:"English",
	discription: "it is international language learning book",
	author:"naveen",
	price: 5,
	pagecount: 1,
	isbn:"",
	id:8,
},
{
	name:"social",
	discription: "it is sample book for children",
	author:"naveen",
	price: 7,
	pagecount: 1,
	isbn:"",
	id:9,
},
{
	name:"telugu",
	discription: "it is language learning book",
	author:"naveen",
	price: 8,
	pagecount: 1,
	isbn:"",
	id:10,
},
{
	name:"English",
	discription: "it is international language learning book",
	author:"naveen",
	price: 5,
	pagecount: 1,
	isbn:"",
	id:11,
}
]