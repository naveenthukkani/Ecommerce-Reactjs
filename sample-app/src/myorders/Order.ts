export interface Order {
    date: String,
    status: String,
    title: String,
    subtitle: String,
    author: String,
    price: number,
    id:number,
    pages: number,
    isbn: string
  };