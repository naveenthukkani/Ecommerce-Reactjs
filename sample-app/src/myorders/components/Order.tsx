import React from 'react';
import "../Style.css"
import {Orders} from "../Orders"
import Book from '../../images/book.jpg'

export const Order: React.FC<Orders> = (props) => {
return (
    <div className="Order-conatiner" key={props.id}>
      <div className="Header-view">
            <div className="Date-string">
                <label>Order Placed:  </label>
                <span>{props.date}</span>
            </div>
            <div className="Status">
                <label>Status:  </label>
                <span>{props.status}</span>
            </div>
      </div>
        <div className="Orders-data-container">
            <img className="Orders-img" src={Book}></img>
            <div className="Data-container">
            <span className="Book-title">{props.bookTitle}</span>
            <div>
              <label>Book Price: </label>
              <span>{props.bookPrice}</span>
            </div>
            <div>
              <label>Author Name: </label>
              <span>{props.authorName}</span>
            </div>
            <div>
              <label>Page Count: </label>
              <span>{props.pageCount}</span>
            </div>
            </div>
        </div>
    </div>
)
};

Order.defaultProps ={
    date: new Date().toLocaleDateString(),
    status:"Delivered",
    bookTitle: "maths",
    authorName: "Naveen",
    bookPrice: 80,
    imageUrl: ""
}