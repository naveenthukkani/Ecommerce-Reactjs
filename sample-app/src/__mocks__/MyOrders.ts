import {Orders} from "../../src/myorders/Orders";

export const OrdersList: Array<Orders> = [
        {
            date: new Date().toLocaleString(),
            status: "Delivered",
            isbn: "9781593275846",
            title: "Eloquent JavaScript, Second Edition",
            subtitle: "A Modern Introduction to Programming",
            author: "Marijn Haverbeke",
            pages: 472,
            price: 500,
            id:1
        },
];