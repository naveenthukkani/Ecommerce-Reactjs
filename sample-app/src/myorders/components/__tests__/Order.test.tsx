import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MyOrder} from '../Order';
import { OrdersList } from '../../../__mocks__/MyOrders';

describe('MyOrderComponent', () =>{
    let props:any;
    beforeEach(()=>{
        props = OrdersList[0]
    });

    it('renders the component correctly', () => {
        const component = shallow(<MyOrder {...props} />);
        expect(toJson(component)).toMatchSnapshot();
      });
});