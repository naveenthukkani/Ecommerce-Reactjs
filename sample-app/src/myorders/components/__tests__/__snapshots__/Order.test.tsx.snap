// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyOrderComponent renders the component correctly 1`] = `
<div
  className="Order-conatiner"
  key="1"
>
  <div
    className="Header-view"
  >
    <div
      className="Date-string"
    >
      <label>
        Order Placed:  
      </label>
      <span>
        7/19/2020, 12:30:27 PM
      </span>
    </div>
    <div
      className="Status"
    >
      <label>
        Status:  
      </label>
      <span>
        Delivered
      </span>
    </div>
  </div>
  <div
    className="Orders-data-container"
  >
    <img
      alt="Book"
      className="Orders-img"
      src="book.jpg"
    />
    <div
      className="Data-container"
    >
      <span
        className="Book-title"
      >
        Eloquent JavaScript, Second Edition
      </span>
      <div>
        <label>
          Book Price: 
        </label>
        <span>
          500
        </span>
      </div>
      <div>
        <label>
          Author Name: 
        </label>
        <span>
          Marijn Haverbeke
        </span>
      </div>
      <div>
        <label>
          Page Count: 
        </label>
        <span>
          472
        </span>
      </div>
    </div>
  </div>
</div>
`;
